### PR TITLE
refactor(agw): unify sentry wrappers from sessiond and mme

### PR DESCRIPTION
**Pull request by waqaraqeel**
## Summary

To be reviewed as diff over #8143.

Both session_manager and oai had separate implementations for Sentry functions. So `lte/gateway/c/core/oai/common/sentry_wrapper.cpp` was a duplicate of `lte/gateway/c/session_manager/SentryWrappers.cpp`. This PR removes duplication and pulls Sentry functions into a Magma common lib.

## Test Plan

Tested with `make build_common`, `make build_session_manager` and `make build_oai`. Also ran relevant tests. Verified that the right libraries were being linked.

### CODE REVIEW

**Comment by themarwhal** [comment]
Adding @amarpad as a reviewer as he is the Magma Cmake expert :D

File: lte/gateway/c/core/oai/CMakeLists.txt [hunk 1]
```diff
@@ -129,6 +129,9 @@ include_directories(${PROJECT_SOURCE_DIR}/include)
 # Add magma libraries
 #############################################################
 
+set(MAGMA_COMMON_INCL_DIR $ENV{C_BUILD}/common)
```
Comment (amarpad) [inline lte/gateway/c/core/oai/CMakeLists.txt]: This is incorrect you should expose this as a target and import the target through the find_package(MAGMA_SENTRY) command, it should bring in the include directories itself. Also note include_directories is kinda retired in CMake the pratice is to use target_include_directories

File: lte/gateway/c/core/oai/test/CMakeLists.txt [hunk 2]
```diff
@@ -13,6 +13,9 @@ set(CMAKE_CXX_STANDARD 14)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS OFF)
 
+set(MAGMA_LIB_DIR $ENV{C_BUILD}/magma_common)
+link_directories(${MAGMA_LIB_DIR}/sentry)
```
Comment (amarpad) [inline lte/gateway/c/core/oai/test/CMakeLists.txt]: Same comment here the transitive dependencies should come from target_link_libraries pulling in the dependencies using find_packages.

File: lte/gateway/Makefile [hunk 3]
```diff
@@ -148,7 +148,7 @@ build_python: stop ## Build Python environment
 build_common: format_common ## Build shared libraries
 	$(call run_cmake, $(C_BUILD)/magma_common, $(MAGMA_ROOT)/orc8r/gateway/c/common, $(COMMON_FLAGS))
 
-build_oai: format_core ## Build OAI
```
Comment (amarpad) [inline lte/gateway/Makefile]: Note that we should not explicitly build_common outside the cmake, this is what I tried to fix with the super build. Please revert this and if something is failing you don't have the right dependencies expressed in CMake.

File: orc8r/gateway/c/common/sentry/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    # Include dir
+    message("Building with sentry!")
+    find_path(SENTRY_INCLUDE_DIR NAMES sentry.h )
+    target_link_libraries(MAGMA_SENTRY ${SENTRY_LIB} MAGMA_CONFIG)
+    set (CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -DSENTRY_ENABLED=1")
+    set (CMAKE_CXX_FLAGS_RELWITHDEBINFO "${CMAKE_CXX_FLAGS_RELWITHDEBINFO} -DSENTRY_ENABLED=1")
+endif()
+
+
+# copy headers to build directory so they can be shared with OAI,
```
Comment (amarpad) [inline orc8r/gateway/c/common/sentry/CMakeLists.txt]: Note OAI shouldn't require this as it uses a super build

File: lte/gateway/Makefile [hunk 5]
```diff
@@ -204,7 +204,7 @@ scan_oai: ## Scan OAI
 test_python: stop ## Run all Python-specific tests
 	make -C $(MAGMA_ROOT)/lte/gateway/python test_all
 
-test_oai: ## Run all OAI-specific tests
+test_oai: build_common ## Run all OAI-specific tests
```
Comment (amarpad) [inline lte/gateway/Makefile]: same comment here

**Review by amarpad — CHANGES_REQUESTED** [review]

**Comment by amarpad** [comment]
thanks for the cleanup btw

File: lte/gateway/c/core/oai/CMakeLists.txt (hunk 1 above)
Comment (waqaraqeel) [reply lte/gateway/c/core/oai/CMakeLists.txt]: Done. `find_package` was key. Much simpler cmake changes. Thanks!

File: lte/gateway/Makefile (hunk 3 above)
Comment (waqaraqeel) [reply lte/gateway/Makefile]: Removed.

File: orc8r/gateway/c/common/sentry/CMakeLists.txt (hunk 4 above)
Comment (waqaraqeel) [reply orc8r/gateway/c/common/sentry/CMakeLists.txt]: Not needed indeed.

File: orc8r/gateway/c/common/sentry/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+  message("Building with sentry!")
+  find_path(SENTRY_INCLUDE_DIR NAMES sentry.h )
+  target_link_libraries(MAGMA_SENTRY ${SENTRY_LIB} MAGMA_CONFIG)
+  set (CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -DSENTRY_ENABLED=1")
+  set (CMAKE_CXX_FLAGS_RELWITHDEBINFO "${CMAKE_CXX_FLAGS_RELWITHDEBINFO} -DSENTRY_ENABLED=1")
+endif()
+
+target_include_directories(MAGMA_SENTRY PUBLIC
+    "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>"
+    $<TARGET_FILE_DIR:MAGMA_SENTRY>)
```
Comment (amarpad) [inline orc8r/gateway/c/common/sentry/CMakeLists.txt]: you need target_file_dir only if you have generated files in the build dir, I don't see them here, so you probably need only the source headers (SOURCE_DIR line).

**Review by amarpad — APPROVED** [review]
One minor nit.

File: lte/gateway/c/core/oai/common/CMakeLists.txt [hunk 7]
```diff
@@ -63,26 +62,11 @@ add_subdirectory(glogwrapper)
...
-      NAMES
-      sentry.h
-      )
-  target_link_libraries(COMMON ${SENTRY_LIB} MAGMA_CONFIG)
-  target_include_directories(COMMON PUBLIC
-      ${CMAKE_CURRENT_SOURCE_DIR}
-      $<TARGET_FILE_DIR:COMMON>
-      )
-endif ()
+target_link_libraries(COMMON
```
Comment (themarwhal) [inline lte/gateway/c/core/oai/common/CMakeLists.txt:66]: @amarpad Did you use a tool to format your CMake changes way back?

File: orc8r/gateway/c/common/sentry/CMakeLists.txt (hunk 6 above)
Comment (waqaraqeel) [reply orc8r/gateway/c/common/sentry/CMakeLists.txt]: Removed. Did not need it indeed.

File: lte/gateway/c/core/oai/common/CMakeLists.txt (hunk 7 above)
Comment (themarwhal) [reply lte/gateway/c/core/oai/common/CMakeLists.txt:66]: OK amar is out this week it seems, but the spacing looks a bit of here? (I can't tell if it's just the GitHub UI though. As long as it's consistent with the rest of the file I'm OK)

**Review by themarwhal — APPROVED** [review]

**Comment by themarwhal** [comment]
Let's get @ardzoht's blessing to cover agw-python and agw-mme approvals.

File: lte/gateway/c/core/oai/common/CMakeLists.txt (hunk 7 above)
Comment (waqaraqeel) [reply lte/gateway/c/core/oai/common/CMakeLists.txt:66]: It is consistent. There are no spacing changes here in fact, just append to a line.

**Review by ardzoht — APPROVED** [review]
lgtm

**Pull request merged**